bin/akula.rs

```diff
@@ -211,6 +236,12 @@ fn main() -> anyhow::Result<()> {
                     file.flush()?;
                 }
 
+                let consens_config = engine_factory(
```

Comment:
why instance a engine_factory new one, could you reuse the same consensus instance?

Reply:
The other is Arc<> type, I used here is Box<> type, if Arc<>type, it need the internal struct implement clone which was not applied in some following structs

Reply:
ok, if the code only misses clone function, you could add Clone derive trait. if you want to use the mutable engine, you cannot use Arc. you could instance engine where you need it.